app/javascript/packages/countdown/countdown-alert-element.ts

```diff
@@ -0,0 +1,33 @@
+import { CountdownElement } from './countdown-element';
+
+export class CountdownAlertElement extends HTMLElement {
+  connectedCallback() {
+    this.setVisibilityTimeout();
+  }
+
+  get showAtRemaining(): number | null {
+    return Number(this.getAttribute('show-at-remaining')) || null;
+  }
+
+  get countdown(): CountdownElement {
+    return this.querySelector('lg-countdown')!;
+  }
+
+  setVisibilityTimeout() {
+    const { showAtRemaining } = this;
+    if (!showAtRemaining) {
+      return;
+    }
+
+    setTimeout(() => this.show(), this.countdown.timeRemaining - showAtRemaining);
```

Comment:
This isn't working correctly, I suspect because the `CountdownElement` isn't "connected" by the time this is being run, so `timeRemaining` is `undefined`.

Reply:
Doing a bit of searching around for how to "wait for" a connected child, I'm getting the sense that these kinds of direct dependencies between the components are typically discouraged. Still, I don't know that it makes a whole lot of sense to try to bake in a "show" behavior to the base countdown component.

Another idea I'd considered previously which might work better and avoid a direct tie between the two components would be to have the countdown element emit a "tick" event each time it updates, then have the wrapper listen for that event. It's a little less efficient to be handling dozens of events vs. scheduling a single timeout, but I don't think it's too troubling since it's spread out over a long period of time.

Reply:
Switched to a new tick event in c6d2dec.